local-tests/chainrunner/chain.py

```diff
@@ -169,14 +169,26 @@ def fork(self, forkoff_path, ws_endpoint, snapshot_file=None):
         chainspec = join(self.path, 'chainspec.json')
         snapshot = snapshot_file if snapshot_file else join(self.path, 'snapshot.json')
         cmd = [check_file(forkoff_path), '--ws-rpc-endpoint', ws_endpoint,
-                '--initial-spec-path', chainspec,
-                '--snapshot-path', snapshot,
-                '--combined-spec-path', forked]
+               '--initial-spec-path', chainspec,
+               '--snapshot-path', snapshot,
+               '--combined-spec-path', forked]
         if snapshot_file:
             cmd.append('--use-snapshot-file')
         subprocess.run(cmd, check=True)
         self.set_chainspec(forked)
 
+    def get_highest_imported(self, nodes=None):
+        """Return the highest block id imported by all the selected nodes"""
+        nodes = nodes or range(len(self.nodes))
```

Comment:
`nodes = range(len(self.nodes)) if nodes is None else nodes`

check what happens if you pass `nodes=[]`

Reply:
We return `-1`. If we strictly interpret the name of the function, it should return "the maximum `k` such that for each `x` in `nodes` `x` imported `k`" and I agree that it's a bit misleading as such `k` doesn't exist. But I find returning `-1` as an acceptable default behavior.

Reply:
check again :P

Reply:
(And I like the idea of returning `-1`, cause it's somewhat consistent with the convention that max of an empty set is minus infinity - here `-1` will be smaller than any possible block height, so you conveniently keep this property of `-inf`. Returning `None` would also be fine.)

Reply:
a yeah, the full range, tricky

Reply:
Actually after some thought I'd rather convert both `None` and the empty list to the full range and describe it in the docstring. I would really want `None` to get converted, and treating `[]` differently here seems unnecessary to me here.

Reply:
`None` gets converted because it's a python convention of providing default value if you can't do this in the method signature.

I don't think converting `[]` to full range is a good idea, as I don't see the use case other than "by accident". If you dislike the idea of returning `-1`, then I'd suggest either returning `None`, or raising an exception.

Reply:
ok, let it be -1